File: android/src/main/java/com/getcapacitor/community/classes/events/FailToEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.community.BridgefyController.Reason;

public class FailToEvent {

    @NonNull
    Reason reason;

    public FailToEvent(@NonNull Reason reason) {
        this.reason = reason;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("type", reason.type());

        if (reason.message() != null) {
            result.put("message", reason.message());
        }

        if (reason.code() != null) {
            result.put("code", reason.code());
        }

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Events/FailToEvent.swift
import Foundation
import Capacitor

@objc public class FailToEvent: NSObject {
    let error: BridgefyError?

    init(_ error: BridgefyError?) {
        self.error = error
    }

    public func toJSObject() -> JSObject {
        var result = JSObject()

        if let error = error {
            result["type"] = error.type

            if let message = error.message {
                if !message.isEmpty {
                    result["message"] = message
                }
            }

            if let code = error.code {
                result["code"] = code
            }
        }

        return result
    }
}
